app/models/github_api.rb

```diff
@@ -35,6 +35,12 @@ def access_token
     @access_token = data['access_token'].first
   end
 
+  def secondary_emails
+    return @secondary_emails if @secondary_emails
+    @secondary_emails = fetch_email_responses.select { |hsh| !hsh['primary'] && hsh['verified'] }
+                                             .map { |response| response['email'] }.compact
+  end
```

Comment:
If this method gets all the emails, it should be named appropriately. **all_emails** or just **emails**.

Reply:
This will return non-primary verified emails. Can we name it as verified_emails ?

Reply:
sounds good

Reply:
We use this method to find the github_api_account. If it doesn't return the primary github email, then we would end up finding only those users whose email on OH is from one of their secondary Github Emails. Is this what we want?

Reply:
We are checking for account with primary email first. If we dont find the account, then we will look for account matching with secondary Github Emails.

>   def github_api_account
    @account ||= Account.find_by(email: github_api.email)
    @account ||= Account.where(email: github_api.secondary_emails).first
  end

Reply:
Thanks for pointing it out. In this case the name secondary_emails makes complete sense.